adafruit_slideshow.py

```diff
@@ -0,0 +1,209 @@
+CircuitPython helper library for displaying a slideshow of images on a board with a built-in
+display.
+
+* Author(s): Kattni Rembor, Carter Nelson
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+ * `Adafruit Hollowing M0 Express <https://www.adafruit.com/product/3900>`_"
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+import time
+import os
+import random
+import board
+import displayio
+import pulseio
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Slideshow.git"
+
+
+class SlideShow():
+    """Class for displaying a slideshow of .bmp images on boards with built-in
+    displays.
+
+    :param str folder: Specify the folder containing the image files, in quotes. Default is
+                       the root directory, ``"/"``.
+
+    :param order: The order in which the images display. You can choose random (``RANDOM``) or
+                  alphabetical (``ALPHA``). Default is ``RANDOM``.
+
+    :param loop: Specify whether to loop the images or play through the list once. `True`
+                 if slideshow will continue to loop, ``False`` if it will play only once.
+                 Default is ``True``.
+
+    :param int dwell: The number of seconds each image displays, in seconds. Default is 3.
+
+    Example code for Hollowing Express. With this example, the slideshow will play through once
+    in alphabetical order:
+
+    .. code-block:: python
+
+        import adafruit_slideshow
+
+        slideshow = adafruit_slideshow.SlideShow()
+        slideshow.loop = False
+        slideshow.order = slideshow.ALPHA
+
+        while slideshow.update():
+            pass
+    """
+    MAX_BRIGHTNESS = 2 ** 15
+
+    ALPHA = 0
+    RANDOM = 1
+
+    FADE_IN = 0
+    SHOW_IMG = 1
+    FADE_OUT = 2
+    LOAD_IMG = 3
+
+    def __init__(self, folder="/", order=RANDOM, loop=True, dwell=3):
+        self._group = displayio.Group()
+        board.DISPLAY.show(self._group)
+        self._backlight = pulseio.PWMOut(board.TFT_BACKLIGHT)
+        self._folder = folder
+        self._order = order
+        self._loop = loop
+        self._dwell = dwell
+        self._current_state = self.LOAD_IMG
+        self._img_start = None
+        self._file_list = None
+        self._images = None
+        self._load_images()
+
+    @property
+    def folder(self):
+        """Specifies the folder containing the image files. Default is the root directory, ``"/"``.
+        """
+        return self._folder
+
+    @folder.setter
+    def folder(self, folder):
+        self._folder = folder
+
+    @property
+    def order(self):
+        """Specifies the order in which the images are displayed. Options are random (``RANDOM``) or
+        alphabetical (``ALPHA``). Default is ``RANDOM``."""
+        return self._order
+
+    @order.setter
+    def order(self, order):
```

Comment:
If there are only 2 options, I would check that the supplied argument meets one of them:
```python
def order(self, order):
    if not order in [ALPHA, RANDOM]:
        raise ValueError("Order must be either 'RANDOM' or 'ALPHA'")
```
This should also be checked in `__init__`.

Reply:
So every instance of `_order` gets changed to `order` then?